pandas/core/internals.py

```diff
@@ -1879,7 +1879,9 @@ def _holder(self):
     @property
     def _can_hold_na(self):
         # The default ExtensionBlock._can_hold_na is True
-        return self._holder._can_hold_na
+        # Needed getattr to pass our old extension tests
+        # Check if geopandas needs this.
+        return getattr(self._holder, '_can_hold_na', True)
```

Comment:
@jorisvandenbossche I needed this to pass the `test_external_block` tests. That uses an `np.ndarray` as the holder, which doesn't technically satisfy our new interface, so `pd.concat()` didn't work. Should I keep this getattr around?

Reply:
It seems we don't have that defined in our GeometryBlock, so I suppose we inherit it from Block.

It would be nice if, for now, geopandas master keeps working with pandas master (although I am certainly planning to use the ExtensionArray interface in geopandas and remove the GeometryBlock (or at least only use that for older pandas versions)). 
But maybe for that, we can also keep the tests for external Block as they were (inheriting from NonConsolidatebleBlock instead of extension block), although not sure if that is related to what you are raising here.

Reply:
So to answer your actual question :-) -> I am not using ExtensionBlock in geopandas but inherit from NonConsolidatableBlock, so maybe this getattr is not needed. 
But I would keep the external block tests I wrote for now, so might (as said above) keep the CustomBlock inheriting from nonconsolidatable as before.